### PR TITLE
fix issue #731

### PR DESCRIPTION
**Pull request by BoleynSu**
fix issue #731

### CODE REVIEW

File: math/src/main/java/breeze/linalg/operators/DenseVectorSupportMethods.java [hunk 1]
```diff
@@ -10,35 +10,32 @@
...
+     * WARNING: only returns the right answer for vectors of length < MAX_SMALL_DOT_PRODUCT_LENGTH
      * @param a
      * @param b
      * @param length
      * @return
      */
     public static double smallDotProduct_Double(double[] a, double[] b, int length) {
-        double sumA = 0.0;
-        double sumB = 0.0;
+        double sum = 0.0;
```
Comment (dlwh) [inline math/src/main/java/breeze/linalg/operators/DenseVectorSupportMethods.java]: please revert this. this code is deliberately written this way to be faster.

File: math/src/main/java/breeze/linalg/operators/DenseVectorSupportMethods.java [hunk 2]
```diff
@@ -87,6 +84,8 @@ public static double dotProduct_Double(double[] a, int aoff, double[] b, int bof
     public static float smallDotProduct_Float(float[] a, float[] b, int length) {
         float sum = 0.0f;
         switch (length) {
+            case 7:
```
Comment (dlwh) [inline math/src/main/java/breeze/linalg/operators/DenseVectorSupportMethods.java]: thanks for catching this

**Comment by dlwh** [comment]
could you add a test case please

File: math/src/main/java/breeze/linalg/operators/DenseVectorSupportMethods.java (hunk 1 above)
Comment (BoleynSu) [reply math/src/main/java/breeze/linalg/operators/DenseVectorSupportMethods.java]: I removed sumA and sumB to make smallDotProduct for both double and float the same. If I revert this, should I use the same optimization for float?

**Comment by dlwh** [comment]
ideally we'd benchmark it, but i think that's a good idea.

Thanks,
David

**Comment by BoleynSu** [comment]
@dlwh I have added tests and reverted sum back to sumA and sumB.

**Comment by dlwh** [comment]
Thanks!

**Pull request merged**